src/openrct2/ride/ShopItem.cpp

```diff
@@ -87,139 +87,55 @@ money32 shop_item_get_common_price(Ride* forRide, int32_t shopItem)
             {
                 continue;
             }
-            if (rideEntry->shop_item == shopItem)
+            if (rideEntry->shop_item[0] == shopItem)
             {
-                return ride.price;
+                return ride.price[0];
             }
-            if (rideEntry->shop_item_secondary == shopItem)
+            if (rideEntry->shop_item[1] == shopItem)
             {
-                return ride.price_secondary;
+                return ride.price[1];
             }
-            if (shop_item_is_photo(shopItem) && (ride.lifecycle_flags & RIDE_LIFECYCLE_ON_RIDE_PHOTO))
+            if (ShopItems[shopItem].IsPhoto() && (ride.lifecycle_flags & RIDE_LIFECYCLE_ON_RIDE_PHOTO))
             {
-                return ride.price_secondary;
+                return ride.price[1];
             }
         }
     }
 
     return MONEY32_UNDEFINED;
 }
 
-bool shop_item_is_photo(int32_t shopItem)
+bool shop_item_has_common_price(int32_t shopItem)
```

Comment:
shopItem can be const
Why not make this a member function?

Reply:
Because shop items do not know their own ID.